include/vast/Dialect/HighLevel/HighLevelCF.td

```diff
@@ -166,12 +166,72 @@ def HighLevel_CondOp
       "builder_callback_ref":$condBuilder,
       "builder_callback_ref":$thenBuilder,
       "builder_callback_ref":$elseBuilder
-    )>
+    ), [{
+        InsertionGuard guard($_builder);
+        build_region($_builder, $_state, condBuilder);
+        build_region($_builder, $_state, thenBuilder);
+        build_region($_builder, $_state, elseBuilder);
+        $_state.addTypes(type);
+    }] >
   ];
 
   let assemblyFormat = [{ attr-dict `:` type(results) $condRegion `?` $thenRegion `:` $elseRegion }];
 }
 
+def HighLevel_CondOp       : HighLevel_CondOpBase< "cond" > {
```

Comment:
Why `HighLevel_CondOpBase` since you dont use anything from the base and override everything (builders, format)?